engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -478,26 +478,36 @@ class WorkflowActor(workflowToStart: WorkflowToStart,
     goto(finalState) using data.copy(currentLifecycleStateActor = None)
   }
 
-  private[workflow] def makeFinalizationActor(workflowDescriptor: EngineWorkflowDescriptor, jobExecutionMap: JobExecutionMap, workflowOutputs: CallOutputs) = {
-    val copyWorkflowOutputsActorProps = stateName match {
+  private[workflow] def makeFinalizationActor(workflowDescriptor: EngineWorkflowDescriptor, jobExecutionMap: JobExecutionMap, workflowOutputs: CallOutputs): ActorRef = {
+    val actorsPropsOptionList: Option[List[Props]] = stateName match {
       case InitializingWorkflowState => None
-      case _ => Option(CopyWorkflowOutputsActor.props(workflowIdForLogging, ioActor, workflowDescriptor, workflowOutputs, stateData.initializationData))
+      case _ => Option(List(CopyWorkflowOutputsActor.props(workflowIdForLogging, ioActor, workflowDescriptor, workflowOutputs, stateData.initializationData),
+        CopyWorkflowMetadataActor.props(workflowIdForLogging, ioActor, serviceRegistryActor, workflowDescriptor, workflowOutputs, stateData.initializationData)))
     }
-    
+
+    val actorsList: List[Props] = actorsPropsOptionList.getOrElse(List(Props.empty))
+
+    val copyWorkflowOutputsActorProps = Option(actorsList(0))
```

Comment:
Consider using `head`, `last`

Reply:
Thanks, I like this proposal, the current view is really clunky